### PR TITLE
feat: add resizeOnImagesReady option

### LINKED ISSUE #396: Add the resizeOnImagesReady option
## Description
Panel sizes can be changed after the image is loaded, and as a carousel component often includes such images I suggest adding a `resizeOnImagesReady` option to resolve that kind of issue.
@egjs/imready can be used as a dependency of Flicking for this feature.

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
#396

## Details
This closes #396

### CODE REVIEW

**Review by mixed — COMMENTED** [review]
I have a few questions.
1. The path of the testcase seems to be set incorrectly. Is it correct?
2. You need to check from a little more edge case. 
    - For example, what does Flicking do when an image(or video) fails to load? And what if I want to load the image again?

File: src/consts.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ export const DEFAULT_OPTIONS: Readonly<FlickingOptions> = {
   isConstantSize: false,
   renderOnlyVisible: false,
   renderExternal: false,
+  resizeOnImagesReady: false,
```
Comment (mixed) [inline src/consts.ts:65]: Are only images available? Isn't video possible?

File: src/Flicking.ts [hunk 2]
```diff
@@ -828,9 +831,25 @@ class Flicking extends Component<{
...
   private listenResize(): void {
-    if (this.options.autoResize) {
+    const options = this.options;
+
+    if (options.autoResize) {
       window.addEventListener("resize", this.resize);
     }
+
+    if (options.resizeOnImagesReady) {
+      const imageReadyChecker = new ImReady();
```
Comment (mixed) [inline src/Flicking.ts:841]: Should you create ImReady every time?

File: src/Flicking.ts [hunk 3]
```diff
@@ -828,9 +831,25 @@ class Flicking extends Component<{
...
 
   private listenResize(): void {
-    if (this.options.autoResize) {
+    const options = this.options;
+
+    if (options.autoResize) {
       window.addEventListener("resize", this.resize);
     }
+
+    if (options.resizeOnImagesReady) {
```
Comment (daybrush) [inline src/Flicking.ts:840]: When destroying before becoming ready, imready should also be removed.

**Comment by WoodNeck** [comment]
@mixed 
I've revised some specs based on your review, please check it again :)

@daybrush
I've found out that `preReady` is not triggered when there're images with `loading="lazy"` only, which breaks my tests.
Is that a real spec of imready?

File: src/Flicking.ts [hunk 4]
```diff
@@ -559,7 +566,11 @@ class Flicking extends Component<{
     const parsedElements = parseElement(element);
 
     const insertingIndex = Math.max(viewport.panelManager.getRange().min - parsedElements.length, 0);
-    return viewport.insert(insertingIndex, parsedElements);
+    const prependedPanels = viewport.insert(insertingIndex, parsedElements);
+
+    this.contentsReadyChecker.check(prependedPanels.map(panel => panel.getElement()));
```
Comment (mixed) [inline src/Flicking.ts:571]: If `resizeOnContentsReady` is false, isn't `this.contentsReadyChecker` undefined? I think that You need to check everywhere you have used `this.contentsReadyChecker` . And the statement is repeated. How about making it a method?

**Review by mixed — COMMENTED** [review]
Thank you. What I suggested seems to have worked well. And If the path is correct, then the coverage is reduced by 60%, so confirmation is required.

**Comment by WoodNeck** [comment]
@mixed 
Well, you can ignore the coverage. I left `only` in the test by mistake. so it will be increased after the test is successful.

File: src/Flicking.ts (hunk 4 above)
Comment (WoodNeck) [reply src/Flicking.ts]: Thanks, I've fixed it.

**Review by mixed — APPROVED** [review]
LGTM!

File: src/Flicking.ts [hunk 5]
```diff
@@ -495,6 +501,7 @@ class Flicking extends Component<{
     }
 
     this.viewport.destroy(option);
+    this.contentsReadyChecker?.off();
```
Comment (daybrush) [inline src/Flicking.ts:504]: It is better to use destroy rather than off.

Comment (WoodNeck) [reply src/Flicking.ts]: I've fixed it, can you check what I've written above?

**Comment by daybrush** [comment]
@WoodNeck 

In imready, all the test codes related to loading have passed, and it works well in UI tests.

File: test/unit/options.spec.ts [hunk 6]
```diff
@@ -1458,4 +1456,241 @@ describe("Initialization", () => {
...
+        Object.defineProperty(imgEl, "complete", { value: false });
+
+        panelEl.appendChild(imgEl);
+        return panelEl;
+      });
+
+      flicking.resize = resizeSpy;
+
+      // When
+      flicking.append(newPanelElements);
```
Comment (daybrush) [inline test/unit/options.spec.ts:1579]: @WoodNeck 

Do you check imready for the appended element?

**Comment by daybrush** [comment]
@WoodNeck 

Would you like to update to version 1.1.0?

**Comment by WoodNeck** [comment]
@daybrush 
I'm using v1.1.0 in this PR

**Comment by daybrush** [comment]
@WoodNeck 

Have you ever changed the timing?
After checking, setTimeout does not occur.

**Review by daybrush — APPROVED** [review]
LGTM

**Pull request merged**